src/Sulu/Bundle/AdminBundle/Resources/js/components/ResourceLocator/ResourceLocator.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React from 'react';
+import Input from '../Input';
+import resourceLocatorStyles from './resourceLocator.scss';
+
+type Props = {
+    value: string,
+    onChange: (value: string) => void,
+    mode: 'full' | 'leaf',
+};
+
+export default class ResourceLocator extends React.PureComponent<Props> {
+    static defaultProps = {
+        value: '/',
+    };
+
+    fixed: string = '';
+    changeable: string = '';
+
+    componentWillMount = () => {
+        const {value, mode} = this.props;
+        let parts;
+
+        switch (mode) {
+            case 'full':
+                this.fixed = '/';
+                this.changeable = value.substring(1);
+                break;
+            case 'leaf':
+                parts = value.split('/');
+                this.changeable = parts.pop();
+                this.fixed = parts.join('/') + '/';
+                break;
+        }
+    };
+
+    componentWillReceiveProps = (nextProps: Props) => {
+        this.changeable = nextProps.value.substring(this.fixed.length);
+    };
+
+    handleChange = (value: string) => {
+        const {onChange} = this.props;
+
+        onChange(this.fixed + value);
+    };
+
+    render() {
+        return (
+            <div className={resourceLocatorStyles.resourceLocatorContainer}>
+                <span className={resourceLocatorStyles.fixed}>{this.fixed}</span>
+                <Input onChange={this.handleChange} type="string" value={this.changeable} />
```

Comment:
`string` is actually not a valid type here... Because `Input` just passes this type to the `input` type. And it seems as if the `Input` prop types are not specific enough here. In this special case I would omit the `type` (because `text` is the default). Maybe we should also adapt the types of `Input`?